public/app/plugins/datasource/grafana-azure-monitor-datasource/utils/migrateQuery.ts

```diff
@@ -146,7 +146,7 @@ function migrateDimensionToResourceObj(query: AzureMonitorQuery): AzureMonitorQu
         resourceGroup: details?.resourceGroup,
         metricNamespace: details?.metricNamespace,
         resourceName: details?.resourceName,
-        resourceUri: undefined,
+        resourceUri: details.subscription ? undefined : query.azureMonitor.resourceUri,
```

Comment:
I added this to maintain the behaviour of setting the `resourceUri` to `undefined`. I'm assuming (it may be a poor assumption) that if the `subscription` value is set then we can unset the `resourceUri`, otherwise we maintain the value so that the migration can be carried out later when `applyTemplateVariables` is called.

The current behaviour of `resourceUri: undefined` means that we remove the template variable value before it is actually applied.

Reply:
I would not remove it if it's defined: If it's defined, it should take precedence over the rest of params (subs, group, name...) and the backend will skip the URI generation.

Reply:
Does this not go against our deprecation of `resourceUri`?

Reply:
You are right, we need to clean up the resourceUri value, since if not, the UI, showing a subs + group + name, won't reflect what is actually being used in the backend (the resource URI).

I was thinking about our conversation in https://github.com/grafana/grafana/issues/55372. If we have a template variable with multiple resources from different resource groups, the more straight forward way to support that is to have a list of resource IDs so supporting both formats sounds like something we will need to do. In any case, it's not something we are prepared to do at the moment.

Going to the original comment here, the condition is probably better to also take into account the resource group and name, since sometimes the subscription is automatically added using the default value from the data source configuration.

```suggestion
        resourceUri: (details.subscription && details?.resourceGroup && details?.resourceName) ? undefined : query.azureMonitor.resourceUri,
```